src/test/tarball-file-system.test.ts

```diff
@@ -0,0 +1,88 @@
+import {describe, it, expect, beforeEach, afterEach} from 'vitest';
+import {TarballFileSystem} from '../tarball-file-system.js';
+import {detectAndPack} from '../detect-and-pack-node.js';
+import * as fs from 'node:fs/promises';
+import * as path from 'node:path';
+import {tmpdir} from 'node:os';
+
+describe('TarballFileSystem', () => {
+  let tempDir: string;
+
+  beforeEach(async () => {
+    tempDir = await fs.mkdtemp(path.join(tmpdir(), 'reporter-test-'));
+  });
+
+  afterEach(async () => {
+    await fs.rm(tempDir, {recursive: true, force: true});
+  });
+
+  describe('fileExists', () => {
+    it('should return false when tsconfig.json does not exist in tarball', async () => {
```

Comment:
these need updating the same as the local ones, and the duplicate tests further down removing just the same

Reply:
these are going to be fixed on #71